### PR TITLE
Wrapper methods around request package methods

### PR DESCRIPTION
**Pull request by michaelknopf**
Created wrappers methods around requests.get/put/post/delete for better code reusability.

Every call to `requests.<method>` has some similar boilerplate code.  Let's get rid of it by adding some wrapper methods.

### CODE REVIEW

**Comment by dsolmirano** [comment]
@manaschoksi  can you review and approve if you agree with the changes

File: trustar/TruStar.py [hunk 1]
```diff
@@ -98,12 +102,14 @@ def normalize_timestamp(date_time):
...
         return datetime_dt.isoformat()
 
-    def get_token(self, verify=True):
+    def get_token(self, verify=None):
         """
         Retrieves the OAUTH token generated by your API key and API secret.
         this function has to be called before any API calls can be made
         :param verify: boolean - ignore verifying the SSL certificate if you set verify to False
         """
+        if verify is None:
```
Comment (manaschoksi) [inline trustar/TruStar.py]: Should we even allow users to pass this per SDK basis ? I think we should remove this altogether and always use self.verify

File: trustar/TruStar.py [hunk 2]
```diff
@@ -134,8 +140,89 @@ def __get_headers(self, is_json=False):
...
+        :param method: The method of the request ("GET", "PUT", "POST", or "DELETE")
+        :param path: The path of the request, i.e. the piece of the URL after the base URL
+        :param headers: A dictionary of headers that will be merged with the base headers for the SDK
+        :param raise_for_status: Whether to raise an exception if the response's status indicates an error
+        :param verify: Whether to use SSL verification
+        :param kwargs: Any extra keyword arguments.  These will be forwarded to requests.request.
+        :return: The response object.
+        """
+
+        if verify is None:
```
Comment (manaschoksi) [inline trustar/TruStar.py]: Yeah, this check looks ugly and no fun ... If we are changing this, can we remove this altogether for per API basis

File: trustar/TruStar.py [hunk 3]
```diff
@@ -134,8 +140,89 @@ def __get_headers(self, is_json=False):
...
+                except Exception:
+                    # if, for some reason, there is no error message in the json, then raise the original exception
+                    raise e
+                # raise a new HTTPError with the proper message
+                raise HTTPError(message, request=e.request, response=e.response)
+
+        return resp
+
+    def get(self, path, **kwargs):
+        """
```
Comment (manaschoksi) [inline trustar/TruStar.py]: docstring :param path: url path and same for kwargs

File: trustar/TruStar.py [hunk 4]
```diff
@@ -134,8 +140,89 @@ def __get_headers(self, is_json=False):
...
+        Convenience method for making GET calls.
+        """
+        return self.request("GET", path, **kwargs)
+
+    def put(self, path, **kwargs):
+        """
```
Comment (manaschoksi) [inline trustar/TruStar.py]: docstring for func params

File: trustar/TruStar.py [hunk 5]
```diff
@@ -134,8 +140,89 @@ def __get_headers(self, is_json=False):
...
+        return self.request("DELETE", path, **kwargs)
+
+    def ping(self, **kwargs):
+        """
+        Ping the API.
+        """
+        return self.get("ping", **kwargs).content.decode('utf-8').strip('\n')
+
+    def version(self, **kwargs):
```
Comment (manaschoksi) [inline trustar/TruStar.py]: docstring for all above functions

File: trustar/TruStar.py [hunk 6]
```diff
@@ -134,8 +140,89 @@ def __get_headers(self, is_json=False):
...
+        """
+        Ping the API.
+        """
+        return self.get("version", **kwargs).content.decode('utf-8').strip('\n')
+
     def get_reports(self, from_time=None, to_time=None, distribution_type=None, submitted_by=None,
-                    enclave_ids=None, verify=True):
+                    enclave_ids=None, **kwargs):
```
Comment (manaschoksi) [inline trustar/TruStar.py]: If kwargs is just done for verify, it does not make sense !!! Its not being used in other places

File: trustar/TruStar.py [hunk 7]
```diff
@@ -148,33 +235,25 @@ def get_reports(self, from_time=None, to_time=None, distribution_type=None, subm
         (by default reports from all enclaves are returned)
         :param verify: Optional server SSL verification, default True
```
Comment (manaschoksi) [inline trustar/TruStar.py]: You don't need this anymore !!

File: trustar/TruStar.py (hunk 1 above)
Comment (michaelknopf) [reply trustar/TruStar.py]: I'm fine with removing this, I see no reason that a user would need to use different values for `verify` on different requests.

Comment (manaschoksi) [reply trustar/TruStar.py]: Yes, lets just remove this. We might need to change the examples, but thats not a big deal.

File: trustar/TruStar.py (hunk 7 above)
Comment (michaelknopf) [reply trustar/TruStar.py]: What don't I need anymore?

Comment (manaschoksi) [reply trustar/TruStar.py]: the docstring :param verify:

**Comment by manaschoksi** [comment]
LGTM ... Do you need to change examples ?

**Comment by michaelknopf** [comment]
Yes, I do!  Working on it, sorry.

**Pull request merged**